src/Symfony/Component/Serializer/DenormalizationResult.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer;
+
+final class DenormalizationResult
+{
+    private $denormalizedValue;
+    private $invariantViolations = [];
+
+    private function __construct()
```

Comment:
Maybe could we use a public constructor with named arguments. It's more idiomatic than named factory methods in PHP8.

Also, this would allow users to access both the normalized value and the errors. It could be useful in some or access the partially denormalized data even in case of error.

Reply:
I disagree: named factories are more expressive about the intention of the caller and thus simplifies validating arguments. As an example, I could improve things here by validating that `$invariantViolations` is not empty in the `failure()` factory. It would not be possible with a single constructor that supports both cases because we lose the intention of the caller: since we don't know if we're in a success or failure case, we can only accept empty arrays in all cases and infer the use case depending on what the variable contains.

We could mitigate this by allowing e.g. `null` for `$invariantViolations`, but then validation would become more complicated and the API less clear.

I added a `$partiallyDenormalizedValue` argument to the `failure()` factory, which is a different name than in `success()`, that would not be possible with a single constructor.